### PR TITLE
Update recline.make

### PR DESCRIPTION
**Pull request by topicus**
Ref: CIVIC-2586
### Recline refactor
- [ ] Modify recline configuration page to allow user set a max file size for recline previews.
- [ ] Provide an update to delete previously used variable.
- [ ] Change current request strategy to use recline backends instead of use jquery ajax calls.
- [ ] Remove global scoped variables since it's a bad practice.
- [ ] Remove two requests strategy.
- [ ] Improve error handling by creating functions to display errors.
- [ ] Fix datastore error message. ([Object Object] message).
- [ ] Add comments to code.
### User stories
- [x] As user I want to preview resources in a grid
- [x] As user I want to preview resources in a chart
- [x] As user I want to preview resources that contains geo data in a map
- [x] As admin I want to limit the size of files that can be previewed
- [x] As admin I want to preview csv files regardless line ending types (\n\r, \r, \n)
- [x] As user I want to see a warning message when resource's file can't be previewed because its format.
- [x] As user I want to see a warning message when resource's file can't be previewed because its size.
- [x] As user I want to see a warning message when a resource file request fail.
- [x] As user I want to see a warning message when a datastore api call fail because resource doesn't exists.
- [x] As user I want to embed recline previews in other pages using an iframe.

### CODE REVIEW

File: recline.js [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        fileSize = Drupal.settings.recline.fileSize;
+        fileType = Drupal.settings.recline.fileType;
+        dataExplorerSettings = {
+          grid: Drupal.settings.recline.grid,
+          graph: Drupal.settings.recline.graph,
+          map: Drupal.settings.recline.map
+        };
+
+        // This is the very basic state collection.
+        state = recline.View.parseQueryString(decodeURIComponent(window.location.hash));
```
Comment (starsinmypockets) [inline recline.js:27]: I know that this isn't realistic for this version of recline but I think hash-routing should ultimately be handled by a router (standard solution)

**Comment by starsinmypockets** [comment]
I did a code review - looks great!
I would like to QA this before merging - is there a strategy for this?

**Comment by dafeder** [comment]
> As user I want to see a warning message when a datastore api call fail because resource doesn't exists.

@topicus not sure why this is in the scope of this PR?

**Comment by topicus** [comment]
@dafeder if a resource it's already imported into the datastore then recline use it instead of the raw file. If fail for any reason a message should be displayed. Do you remember the Error: [Object object] in the recline preview? This should improve that to show a better error message.

**Comment by dafeder** [comment]
@topicus: I see. Here is a screenshot of what happens when I remove the file. I don't see a warning message.

**Comment by topicus** [comment]
@dafeder I've updated this and now it displays the error properly. However it's only forwarding the requests status which is the message "error". 
Anyways I've created another PR to improve that in the okfn's csv.js repo. https://github.com/okfn/csv.js/pull/17

I don't think we need to wait that PR since it's now an error that will pop up frequently and even if it does a message showing it is displayed.

**Comment by dafeder** [comment]
See NuCivic/dkan_datastore#66

**Pull request merged**